openproblems/tasks/denoising/methods/alra.py

```diff
@@ -0,0 +1,54 @@
+from ....tools.decorators import method
+from ....tools.utils import check_version
+
+import numpy as np
+import scprep
+
+_alra = scprep.run.RFunction(
+    setup="""
+    library(SingleCellExperiment)
+    library(Matrix)
+    library(rsvd)
+    source('https://raw.githubusercontent.com/KlugerLab/ALRA/master/alra.R')
+    
+    """,
+    args="sce",
+    body="""
+    assay(sce, "X") <- as(assay(sce, "X"), "CsparseMatrix")
+    reducedDim(sce, "train") <- as(reducedDim(sce, "train"), "CsparseMatrix")
+    completed <- alra(as.matrix(reducedDim(sce, "train")))
+    normCompleted <- completed[[3]]
+    reducedDim(sce, "train") <- as.matrix(normCompleted)
+    sce
+    """,
+)
+
+@method(
+    method_name="ALRA",
+    paper_name="Zero-preserving imputation of scRNA-seq data using low-rank approximation",
+    paper_url="https://www.biorxiv.org/content/10.1101/397588v1",
+    paper_year=2018,
+    code_url="https://github.com/KlugerLab/ALRA",
+    code_version=check_version("scprep"),
+    image="openproblems-r-extras",
+)
+#support method alra_calc takes adata and any arguments to alra
+def alra_calc(adata):
```

Comment:
What is this function doing? Looks like we don't need it, alra is computed in _alra and r_alra doesn't exist.

Reply:
Correct--This was just a version control issue. Re-committing with this function removed.